src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetLogService.java

```diff
@@ -0,0 +1,17 @@
+package eu.europeana.metis.sandbox.service.dataset;
+
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.dto.report.DatasetLogDto;
+import java.util.List;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface DatasetLogService {
+
+  @Transactional
+  Void log(String datasetId, Status status, String message, Throwable e);
```

Comment:
Why did you decide to use `Void` instead of the primitive `void`?
If you were to implement with the primitive `void` it would be consistent with other services

Reply:
I done it cause the method need to be compatible with functional interface:  Function<Throwable, ? extends T> used by exceptionally method. I could change it to void but in any place where I pass it to the exceptionally method, I need to use more complicated lambda:
Instead of
.exceptionally(e -> datasetLogService.logException(createdDatasetId, e));
I need to use
.exceptionally(e -> {
                            datasetLogService.logException(createdDatasetId, e);
                            return null;
                         });